src/variables/components/TypeAheadVariableDropdown.test.tsx

```diff
@@ -87,50 +93,53 @@ describe('Dashboards.Components.VariablesControlBar.TypeAheadVariableDropdown',
     it('renders dropdown with keys as dropdown items', () => {
       const {getByTestId, getAllByTestId} = renderWithRedux(
         <TypeAheadVariableDropdown variableID="03cbdc8a53a63000" />,
-        setInitialState
+        initialState
       )
 
-      const dropdownButton = getByTestId('variable-dropdown--button')
+      const dropdownButton = getByTestId('typeAhead-dropdown--button')
       fireEvent.click(dropdownButton)
-      const dropdownItems = getAllByTestId('variable-dropdown--item').map(
+      const dropdownItems = getAllByTestId('typeAhead-dropdown--item').map(
         node => node.id
       )
 
-      expect(dropdownItems).toEqual(Object.keys(values))
+      expect(dropdownItems).toEqual(Object.keys(xkcd_name_to_url_map))
     })
   })
 
   it('filters properly while typing in the input', () => {
     const {getByTestId, getAllByTestId} = renderWithRedux(
       <TypeAheadVariableDropdown variableID="03cbdc8a53a63000" />,
-      setInitialState
+      initialState
     )
 
     const filterInput = getByTestId(
-      'variable-dropdown-input-typeAhead--map_buckets'
+      `variable-dropdown--${variable_name}--typeAhead-input`
     )
 
     const checkDropdown = (filterText, expectedList) => {
+      fireEvent.click(filterInput)
       fireEvent.change(filterInput, {target: {value: filterText}})
-      const dropdownItems = getAllByTestId('variable-dropdown--item').map(
+      const dropdownItems = getAllByTestId('typeAhead-dropdown--item').map(
         node => node.id
       )
       expect(dropdownItems).toEqual(Object.keys(expectedList))
     }
 
-    // filter on the string 'f':
-    checkDropdown('f', fvalues)
+    // filter on the string 'a':
+    checkDropdown('a', values_with_a)
 
     // clear input, should see everything:
-    checkDropdown('', values)
+    checkDropdown('', xkcd_name_to_url_map)
 
     // filter again by text:
-    checkDropdown('e', evalues)
-    checkDropdown('al', alvalues)
+    checkDropdown('b', values_with_b)
+    checkDropdown('c', values_with_c)
```

Comment:
Since this refactors the existing test this is fine. But I think we'll want to add some cases for searching more than a single letter to test that that things correctly show up (e.g. searching for `ao` to make sure that `Bad Code` shows up) and that things don't show up (e.g. searching for `azzzzzz` to make sure nothing shows up)

Reply:
Got it! I'll add it to the test!